lib/routemap.c

```diff
@@ -1542,18 +1560,23 @@ route_map_result_t route_map_apply(struct route_map *map,
 	int ret = 0;
 	struct route_map_index *index;
 	struct route_map_rule *set;
+	char buf[PREFIX_STRLEN];
+	char buf2[PREFIX_STRLEN];
 
 	if (recursion > RMAP_RECURSION_LIMIT) {
 		flog_warn(
 			EC_LIB_RMAP_RECURSION_LIMIT,
 			"route-map recursion limit (%d) reached, discarding route",
 			RMAP_RECURSION_LIMIT);
 		recursion = 0;
-		return RMAP_DENYMATCH;
+		ret = RMAP_DENYMATCH;
```

Comment:
Why do we need to have another debug here?  Isn't the flog_warn enough?  I say just leave the return RMAP_DENYMATCH

Reply:
Fixed